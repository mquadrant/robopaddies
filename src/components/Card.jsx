import React from 'react'

export default function Card({name,email}) {
    return (
        <div className="card flex-none inline-block bg-green-200 shadow-md p-3 m-2 rounded-lg">
            <img src={`https://robohash.org/${name}?size=200x200`} alt={name}/>
            <div className="my-2">
                <h4 className="font-semibold text-base text-sm text-center">{name}</h4>
                <p className="text-center text-xs">{email}</p>
            </div>
        </div>
    )
}
