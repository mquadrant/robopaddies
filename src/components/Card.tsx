import React from 'react'

interface ICard {
    name: string;
    email: string;
    key?:number;
    id?:number
}

export default function Card({name,email}:ICard) {
    return (
        <div className="card flex-none inline-block bg-green-200 shadow-md p-3 m-2 rounded-lg">
            <img src={`https://robohash.org/${name}?size=200x200`} alt={name}/>
            <div className="my-2">
                <h4 className="font-semibold text-base text-sm text-center">{name}</h4>
                <p className="text-center text-xs">{email}</p>
            </div>
        </div>
    )
}
