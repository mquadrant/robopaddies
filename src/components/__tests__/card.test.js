import React from 'react';
import { shallow} from 'enzyme';
import Card from '../Card';

it('expect to render the Card component', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper.length).toEqual(1);
  });

it('expect to render Card component with snapshot', () => {
    expect(shallow(<Card />)).toMatchSnapshot();
  });


// it('expect to render Card component with snapshot', () => {
//     const wrapper = shallow(<Card />);
//     wrapper.find('[id="counter"]').simulate('click');
//     expect(wrapper.state()).toEqual({count:1});
//   });

