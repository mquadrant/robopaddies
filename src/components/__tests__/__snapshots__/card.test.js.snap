// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component with snapshot 1`] = `
<div
  className="card flex-none inline-block bg-green-200 shadow-md p-3 m-2 rounded-lg"
>
  <img
    src="https://robohash.org/undefined?size=200x200"
  />
  <div
    className="my-2"
  >
    <h4
      className="font-semibold text-base text-sm text-center"
    />
    <p
      className="text-center text-xs"
    />
  </div>
</div>
`;
